### PR TITLE
Implemented cross-clock synchronization for hardware timestamping.

### PR DESCRIPTION
**Pull request by davidv1992**

### CODE REVIEW

File: statime-linux/src/clock/mod.rs [hunk 1]
```diff
@@ -125,7 +117,8 @@ impl Clock for LinuxClock {
         };
 
         if time_properties.is_ptp() {
-            self.clock.set_leap_seconds(leap_indicator)?;
+            // Always do this on the system clock
+            UnixClock::CLOCK_REALTIME.set_leap_seconds(leap_indicator)?;
```
Comment (folkertdev) [inline statime-linux/src/clock/mod.rs:121]: so if self is not CLOCK_REALTIME, what actually happens? we synchronize the system clock from the hardware clock, so what effect does setting the leap second indicator have?

File: statime-linux/src/main.rs [hunk 2]
```diff
@@ -300,6 +422,21 @@ async fn run(
 
         instance.bmca(&mut mut_bmca_ports);
 
+        let mut clock_states: Vec<_> = internal_sync_senders
+            .iter()
+            .map(|_| ClockSyncMode::FromSystem)
+            .collect();
```
Comment (folkertdev) [inline statime-linux/src/main.rs:428]: this is just a repeat right? I think `vec![ClockSyncMode::FromSystem; internal_sync_senders.len()]` would work then?

File: statime-linux/src/main.rs [hunk 3]
```diff
@@ -316,7 +453,7 @@ type BmcaPort = Port<InBmca<'static>, StdRng, LinuxClock, BasicFilter>;
 // It will then move the port into the running state, and process actions. When
 // the task is notified of a BMCA, it will stop running, move the port into the
 // bmca state, and send it on its Sender
-async fn port_task<A: NetworkAddress + PtpTargetAddress>(
+async fn port_task<A: NetworkAddress + PtpTargetAddress + std::fmt::Debug>(
```
Comment (folkertdev) [inline statime-linux/src/main.rs:456]: is this `impl Debug` intentional (I know you wanted to remove it in some places, maybe this was missed? or maybe it's actually useful here)

File: statime-linux/src/clock/mod.rs (hunk 1 above)
Comment (davidv1992) [reply statime-linux/src/clock/mod.rs]: Lets ignore leap seconds for now

File: statime-linux/src/main.rs (hunk 3 above)
Comment (davidv1992) [reply statime-linux/src/main.rs]: nope, removed

**Review by folkertdev — APPROVED** [review]

**Review by folkertdev — APPROVED** [review]

**Pull request merged**